kornia/filters/gaussian.py

```diff
@@ -3,8 +3,59 @@
 import torch
 import torch.nn as nn
 
+from kornia.testing import KORNIA_CHECK_IS_TENSOR
+
 from .filter import filter2d, filter2d_separable
-from .kernels import get_gaussian_kernel1d, get_gaussian_kernel2d
+from .kernels import get_gaussian_kernel1d, get_gaussian_kernel2d, get_n_gaussian_kernel1d, get_n_gaussian_kernel2d
+
+
+def random_gaussian_blur2d(
```

Comment:
I don’t think we want to mix operations that involves random states in the Kornia core functionality. The main application for random states for us data augmentation, and for this it should go to the dedicated module. However, another discussion is whether to provide free functions — in the past we had, but we end up deprecating because of the overhead to maintain and not fully fitting in the full provided features from the modules api e.g. return transforms. So, I’d everything that involves random goes to Kornia.augmentation and in the form of a module. /cc @shijianjian

Reply:
It's really bad named function, sorry about that I refractored the code several times. There is no random logic but it is the same function as `gaussian_blur2d`, however it takes a sigma tensor that could be of shape `Bx2` instead of float. Based on one other comment u made it could probably be only one function instead of one taking in float and the other one taking in a tensor.